src/configureWorkspace/configurePython.ts

```diff
@@ -3,68 +3,224 @@
+    const launchFilePrompt = defaultLaunchFile.get(projectType);
+
+    const opt: vscode.InputBoxOptions = {
+        placeHolder: launchFilePrompt.defaultFile,
+        prompt: launchFilePrompt.prompt,
+        value: launchFilePrompt.defaultFile,
+        validateInput: (value: string): string | undefined => { return value && value.trim().length > 0 ? undefined : "Enter a valid Python file path/module." }
+    };
+
+    const file = await ext.ui.showInputBox(opt);
+
+    // If the input has the .py extension, then assume it is a file.
+    // TODO: is there a more robust way to check?
+    if (file.toLowerCase().endsWith(".py")) {
+        return { file: file.replace(/\\/g, '/') };
+    } else {
+        return { module: file};
+    }
+}
+
+export async function scaffoldPython(context: ScaffolderContext): Promise<ScaffoldFile[]> {
+    const properties: TelemetryProperties & ConfigureTelemetryProperties = context.telemetry.properties;
+    const serviceName = context.folder.name;
+    const rootFolderPath: string = context.rootFolder;
+    const outputFolder = context.outputFolder ?? rootFolderPath;
+
+    const generateComposeFiles = await context.captureStep("compose", quickPickGenerateComposeFiles)();
+    const projectType = getPythonProjectType(context.platform);
+
+    const defaultPort = PythonDefaultPorts.get(projectType);
+    let ports = [];
+
+    if (defaultPort) {
+        ports = await context.promptForPorts([ defaultPort ]);
+    }
+
+    const launchFile = await context.captureStep("pythonFile", promptForLaunchFile)(projectType);
+    const dockerFileContents = genDockerFile(serviceName, launchFile, projectType, ports);
+
+    const files: ScaffoldFile[] = [
+        { fileName: 'Dockerfile', contents: dockerFileContents, open: true },
+        { fileName: '.dockerignore', contents: genCommonDockerIgnoreFile(context.platform) },
+        { fileName: 'requirements.txt', contents: genRequirementsFile(projectType) }
+    ];
+
+    if (generateComposeFiles) {
+        properties.orchestration = 'docker-compose';
+
+        const dockerComposeFile = genDockerCompose(serviceName, ports);
+        const dockerComposeDebugFile = genDockerComposeDebug(serviceName, projectType, ports, launchFile);
+
+        files.push(
+            { fileName: 'docker-compose.yml', contents: dockerComposeFile },
+            { fileName: 'docker-compose.debug.yml', contents: dockerComposeDebugFile });
+    }
+
+    files.forEach(file => {
+    // Remove multiple empty lines with single empty lines, as might be produced
+    // if $expose_statements$ or another template variable is an empty string.
+        file.contents = file.contents
+            .replace(/(\r\n){3,4}/g, "\r\n\r\n")
+            .replace(/(\n){3,4}/g, "\n\n");
+    });
+
+    if (context.initializeForDebugging) {
+        await initializeForDebugging(context, path.join(outputFolder, "Dockerfile"), ports, generateComposeFiles, launchFile, projectType);
```

Comment:
Nit: (here and elsewhere) prefer single-quotes (`'`) to double-quotes (`"`) for strings.